### PR TITLE
Fix formatting check script issue

### PR DESCRIPTION
**Pull request by PopFlamingo**
Fix formatting options and improve formatting script

### Motivation:

An option not yet supported by swift format as of the 0.44.6 version was present in the `.swiftformat` file, which prevented `validate_format.sh` from properly checking that code formatting validity.

### Modifications:

- Remove unsupported `--disable redundantType` option (doesn't exist yet in swiftformat v0.44.6)
- Don't ignore invalid swiftformat invocations
- ~~Prevent validate_format.sh from actually modifying the code~~ (edit: reverted)
- Modify formatting according to the rules specified in `.swiftformat`

### Result:
Fixed formatting options and improved formatting script

### CODE REVIEW

File: Sources/DistributedActorsTestKit/ActorTestKit.swift [hunk 1]
```diff
@@ -81,7 +81,8 @@ extension ActorTestKit {
             name = ActorTestProbe<Message>.naming.makeName(&self.namingContext)
         }
 
-        return ActorTestProbe({ probeBehavior in
+        return ActorTestProbe(
+            { probeBehavior in
```
Comment (ktoso) [inline Sources/DistributedActorsTestKit/ActorTestKit.swift:85]: hate it when swift format messes with those... is there some setting to prevent it?

**Review by ktoso — APPROVED** [review]
Great, thanks :) 

// Discussed more on chat.

**Comment by ktoso** [comment]
CI will fail but soundness should pass now I think.

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-server-bot add to allowlist

File: scripts/validate_format.sh [hunk 2]
```diff
@@ -19,12 +19,13 @@ set -u
...
-SECOND_OUT="$(git status --porcelain)"
-if [[ "$FIRST_OUT" != "$SECOND_OUT" ]]; then
+# check wether the config is ok
+swiftformat --dryrun . >/dev/null || { echo >&2 "'swiftformat' invocation failed."; exit 1; }
+
+# check wether the code is formatted correctly
+swiftformat . --lint
+if [ $? -ne 0 ]; then
   printf "\033[0;31mformatting issues!\033[0m\n"
-  git --no-pager diff
```
Comment (ktoso) [inline scripts/validate_format.sh:27]: please bring this one back tho, we do want to modify the files and print what changed

Comment (PopFlamingo) [reply scripts/validate_format.sh]: @ktoso Done!

File: Package.swift [hunk 3]
```diff
@@ -109,10 +109,10 @@ var targets: [PackageDescription.Target] = [
             "DistributedActorsTestKit",
         ],
         exclude: [
-          "DocumentationProtos/",
+            "DocumentationProtos/",
         ],
         plugins: [
-            "DistributedActorsGeneratorPlugin"
+            "DistributedActorsGeneratorPlugin",
```
Comment (PopFlamingo) [inline Package.swift:115]: @ktoso Are trailing comas fine?

Comment (PopFlamingo) [reply Package.swift:115]: I believe there should be an option to disable this if needed

Comment (ktoso) [reply Package.swift:115]: Yeah that's fine, looking good thanks

File: Sources/DistributedActors/Behaviors.swift [hunk 4]
```diff
@@ -296,7 +298,7 @@ extension _Behavior {
     }
 
     public func _receiveSignalAsync(
-        _ handle: @escaping @Sendable (_ActorContext<Message>, Signal) async throws -> _Behavior<Message>
+        _ handle: @escaping @Sendable(_ActorContext<Message>, Signal) async throws -> _Behavior<Message>
```
Comment (ktoso) [inline Sources/DistributedActors/Behaviors.swift:301]: Sigh swift-format man... always doing something weird 😉

**Review by ktoso — APPROVED** [review]
Thanks!

**Pull request merged**